snuba/datasets/entity.py

```diff
@@ -139,22 +137,64 @@ def build_match(
 
         if self._required_time_column:
             match = build_match(
-                self._required_time_column,
-                [
-                    ConditionFunctions.EQ,
-                    ConditionFunctions.GT,
-                    ConditionFunctions.GTE,
-                    ConditionFunctions.LT,
-                    ConditionFunctions.LTE,
-                ],
-                datetime,
+                self._required_time_column, [ConditionFunctions.EQ], datetime,
             )
             found = any(match.match(cond) for cond in top_level)
-            if not found:
+            if found:
+                return True
+
+            lower, upper = get_time_range_expressions(
+                top_level, self._required_time_column, alias
+            )
+            if not lower or not upper:
                 return False
 
+            # At this point we have valid conditions. However we need to align them and
+            # make sure they don't exceed the max_days. Replace the conditions.
+            self._replace_time_condition(query, *lower, *upper)
+
```

Comment:
If there's a way to simplify this I'm open to suggestions. I have already abstracted this for everything except the time conditions. The time conditions require separate handling from all other conditions so it makes this function seem bloated.

I'll split the required columns and required time columns into separate functions so that the logic is a little simpler to follow. Otherwise I'm not sure how to simplify this.

Reply:
I would just have a validator class for each type of validation that takes query and deither returns noothing or raise an exception. Just as we do for signature validation. Each entity defines its sequence of validators to apply, or is there a reason why this is not possible?
We would not be able to do everything in one pass, but that seems a minor problem.

Also, I am not suggesting to add that t this PR

Reply:
That's definitely something we can tackle. I might actually do that in this PR because I don't like putting all this code into the base entity class.

Reply:
I will refactor this in a separate PR. I have a plan that I think will simplify this.